doc/source/WorkingWithTheNodeGraph/Context/index.md

```diff
@@ -0,0 +1,331 @@
+
+Let's connect the Text node before the ContextVariables node, then select it.
+
+```eval_rst
+.. image:: images/conceptContextFlow2.png
+    :width: 100%
+    :alt: The same network, but with the Text node connected upstream of the ContextVariables node. The Text node is selected, and the Text node still cannot read the Context Variable.
+```
+
+Once again, no luck. The Text node is being called to process by the editors focused on it, but its Text plug doesn't have a value. Clearly, the Text node isn't reaching down the graph to grab `message` from downstream, either.
+What happens if we select the ContextVariables node?
+
+```eval_rst
+.. image:: images/conceptContextFlow3.png
+    :width: 100%
+    :alt: The adjusted network again, but with the Context Variables node selected. Now the Text node can read the Context Variable, and its text is visible in the Viewer.
+```
+
+Success! Now the Text plug is expanding the Context Variable, and the object has the correct resulting geometry. Clearly, Contexts do in fact move downstream.
+
+But why did we need to select the downstream node? Remember, Contexts are created for the called node, meaning we need the ContextVariables node to process before the Text node, so it can insert `message` into the Context before passing it upstream. Since focusing an editor on a node calls on that node to process, selecting that node gives us the result we want in each editor.
+
+For Expression nodes, the same rules apply. Despite Expression nodes often being positioned to one side in the GraphEditor, they are still upstream with regards to processing:  nodes pull values from them, and therefore pass Contexts up to them. While a single Expression node can drive any number of downstream nodes, it will run separately for each unique downstream Context. Take, for example, the following graph:
+
+The Text plugs of both the Text nodes are driven by a single Expression node. Despite the expression code making calls to both Context Variables, it only substitutes the one which is present in the branch that is currently processing. When the left branch is processed, the expression can only successfully read `apples`. When the right branch is processed, the expression can only read `oranges`.
+
+```eval_rst
+.. image:: images/conceptContextExpressionSharing.png
+    :width: 100%
+    :alt: A network where two branches with respective Context Variables both pull values from a shared Expression node. Neither network can access the other's Context Variable.
+```
+
+
+### Globals and locations ###
+
+When processing a 3D scene, there is an exception made for the `scene:path` Context Variable. To save on performance, nodes that generate globals, such as StandardOptions and CustomOptions, will have `scene:path` stripped from their Contexts before processing. They will still pass `scene:path` along to upstream nodes, but their plugs cannot be driven by it.
```

Comment:
See https://docs.google.com/document/d/15mKlkKX3ApvGhgUodfnO6-5yaQ6YgkxS_ijR5Q-n1jQ/edit?disco=AAAAJUGV6Cw

Reply:
Thats a good improvement, thanks. I think it might be backwards though. I think the meaningless part is more important than the performance aspect though. It's not just that it is omitted, the key is that it isn't even logically definable to start with. 

We omit it in cases where someone happens to query such a plug as part of the compute for a location, for performance reasons, but that is secondary to the incongruity of specifying a location for something that is defined as being location invariant.

Reply:
de816fa1978b494bb573b5c12e8377c869a2be77

Reply:
de816fa1978b494bb573b5c12e8377c869a2be77